comms/src/parse/RequestQueue.ts

```diff
@@ -0,0 +1,127 @@
+import { Duplex } from "stream";
+import { Request, requestTable, requestBuilderTable, RequestBuilder } from "./Requests"
+import { Instruction } from "../debug/Instructions";
+
+type PromiseResolver<R> = (value: R | PromiseLike<R>) => void;
+
+export class RequestQueue {
+    private readonly delimiter: string;
+    private stack: string[];
+    private requests: [Request<any>, PromiseResolver<any>][];
+    private catchAllHandler: (line: string) => void;
+
+    private channel: Duplex;
+
+    constructor(channel: Duplex, delimiter: string) {
+        this.delimiter = delimiter;
+        this.stack = [];
+        this.requests = [];
+        this.channel = channel;
+        this.catchAllHandler = (line: string) => {
+            return this.catchAllLogger(line)
+        }
+        this.registerChannelCallbacks();
+    }
+
+    public sendRequest<R>(request: Request<R>): Promise<R> {
```

Comment:
I think this can be made private.
Developers using the package would ideally only see one 'send' method (if we export this class).

Reply:
Sure.